packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -538,6 +539,86 @@ export class ConnectionService {
     }
   }
 
+  /**
+   * If knownConnectionId is passed, it will compare the incoming connection id with the knownConnectionId, and skip the other validation.
+   *
+   * If no known connection id is passed, it asserts that the incoming message is in response to an attached request message to an out of band invitation.
+   * If is the case, and the state of the out of band record is still await response, the state will be updated to done
+   *
+   */
+  public async matchIncomingMessageToRequestMessageInOutOfBandExchange(
+    messageContext: InboundMessageContext,
+    { knownConnectionId }: { knownConnectionId?: string }
```

Comment:
Maybe it should be 

```suggestion
    { expectedConnectionId }: { expectedConnectionId?: string }
```

Reply:
Yeah that makes sense. I was thinking if it is undefined it doesn't make sense. But after thining about it again, if it's undefined we don't really expect a specific connection Id